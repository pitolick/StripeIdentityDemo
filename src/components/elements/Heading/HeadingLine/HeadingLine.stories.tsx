import React from 'react'
import { Meta, StoryObj } from '@storybook/react'
import { HeadingLine } from './HeadingLine'

const meta: Meta<typeof HeadingLine> = {
  title: 'elements/Heading/HeadingLine',
  component: HeadingLine,
  tags: ['autodocs'],
}

export default meta
type Story = StoryObj<typeof HeadingLine>

/**
 * デフォルト
 */
export const Default: Story = {
  args: {
    children: '見出し',
  },
}

/**
 * h1
 */
export const H1: Story = {
  args: {
    children: '見出し',
    tag: 'h1',
  },
}

/**
 * h2
 */
export const H2: Story = {
  args: {
    children: '見出し',
    tag: 'h2',
  },
}

/**
 * h3
 */
export const H3: Story = {
  args: {
    children: '見出し',
    tag: 'h3',
  },
}

/**
 * h4
 */
export const H4: Story = {
  args: {
    children: '見出し',
    tag: 'h4',
  },
}

/**
 * h5
 */
export const H5: Story = {
  args: {
    children: '見出し',
    tag: 'h5',
  },
}

/**
 * h6
 */
export const H6: Story = {
  args: {
    children: '見出し',
    tag: 'h6',
  },
}

/**
 * カスタムクラス
 */
export const CustomClass: Story = {
  args: {
    children: '見出し',
    className: 'text-red',
  },
}

/**
 * 子要素
 */
export const Children: Story = {
  args: {
    children: (
      <>
        <span className="text-4xl">これは見出しです。</span>
        <br />
        <span className="text-xl">これはサブタイトルです。</span>
      </>
    ),
  },
}
