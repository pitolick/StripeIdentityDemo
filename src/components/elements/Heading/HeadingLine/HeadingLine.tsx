/**
 * 下線付きの見出し
 * @module HeadingLine
 * @example
 *  <HeadingLine>見出し</HeadingLine>
 */
import React from 'react'

/**
 * 下線付きの見出しコンポーネントのプロパティ
 */
interface HeadingLineProps {
  /**
   * 子要素
   */
  children: React.ReactNode
  /**
   * タグ名
   */
  tag?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
  /**
   * class名
   */
  className?: React.HTMLAttributes<HTMLHeadingElement>['className']
}

/**
 * 下線付きの見出しコンポーネント
 */
export const HeadingLine: React.FC<HeadingLineProps> = ({ children, tag: CustomTag = 'h2', className }) => {
  return (
    <>
      <CustomTag
        className={`${className} relative pb-4 font-bold before:absolute before:bottom-0 before:block before:w-full before:border-b-4 before:border-solid before:border-gray-500 before:content-[''] after:absolute after:bottom-0 after:block after:w-1/6 after:border-b-4 after:border-solid after:border-gray-900 after:content-['']`}
      >
        <span className="relative">{children}</span>
      </CustomTag>
    </>
  )
}
