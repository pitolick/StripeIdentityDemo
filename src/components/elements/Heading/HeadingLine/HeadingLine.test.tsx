import React from 'react'
import { render } from '@testing-library/react'
import { HeadingLine } from './HeadingLine'

describe('HeadingLine', () => {
  it('子要素が正しく表示されること', () => {
    const { container } = render(<HeadingLine>見出し</HeadingLine>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('h1タグが正しく設定されること', () => {
    const { container } = render(<HeadingLine tag="h1">見出し</HeadingLine>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('h2タグが正しく設定されること', () => {
    const { container } = render(<HeadingLine tag="h2">見出し</HeadingLine>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('h3タグが正しく設定されること', () => {
    const { container } = render(<HeadingLine tag="h3">見出し</HeadingLine>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('h4タグが正しく設定されること', () => {
    const { container } = render(<HeadingLine tag="h4">見出し</HeadingLine>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('h5タグが正しく設定されること', () => {
    const { container } = render(<HeadingLine tag="h5">見出し</HeadingLine>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('h6タグが正しく設定されること', () => {
    const { container } = render(<HeadingLine tag="h6">見出し</HeadingLine>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('class名が正しく設定されること', () => {
    const { container } = render(<HeadingLine className="text-red">見出し</HeadingLine>)
    expect(container.firstChild).toMatchSnapshot()
  })
})
