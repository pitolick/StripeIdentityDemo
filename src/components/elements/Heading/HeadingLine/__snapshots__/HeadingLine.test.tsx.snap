// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadingLine class名が正しく設定されること 1`] = `
<h2
  class="text-red relative pb-4 font-bold before:absolute before:bottom-0 before:block before:w-full before:border-b-4 before:border-solid before:border-gray-500 before:content-[''] after:absolute after:bottom-0 after:block after:w-1/6 after:border-b-4 after:border-solid after:border-gray-900 after:content-['']"
>
  <span
    class="relative"
  >
    見出し
  </span>
</h2>
`;

exports[`HeadingLine h1タグが正しく設定されること 1`] = `
<h1
  class="undefined relative pb-4 font-bold before:absolute before:bottom-0 before:block before:w-full before:border-b-4 before:border-solid before:border-gray-500 before:content-[''] after:absolute after:bottom-0 after:block after:w-1/6 after:border-b-4 after:border-solid after:border-gray-900 after:content-['']"
>
  <span
    class="relative"
  >
    見出し
  </span>
</h1>
`;

exports[`HeadingLine h2タグが正しく設定されること 1`] = `
<h2
  class="undefined relative pb-4 font-bold before:absolute before:bottom-0 before:block before:w-full before:border-b-4 before:border-solid before:border-gray-500 before:content-[''] after:absolute after:bottom-0 after:block after:w-1/6 after:border-b-4 after:border-solid after:border-gray-900 after:content-['']"
>
  <span
    class="relative"
  >
    見出し
  </span>
</h2>
`;

exports[`HeadingLine h3タグが正しく設定されること 1`] = `
<h3
  class="undefined relative pb-4 font-bold before:absolute before:bottom-0 before:block before:w-full before:border-b-4 before:border-solid before:border-gray-500 before:content-[''] after:absolute after:bottom-0 after:block after:w-1/6 after:border-b-4 after:border-solid after:border-gray-900 after:content-['']"
>
  <span
    class="relative"
  >
    見出し
  </span>
</h3>
`;

exports[`HeadingLine h4タグが正しく設定されること 1`] = `
<h4
  class="undefined relative pb-4 font-bold before:absolute before:bottom-0 before:block before:w-full before:border-b-4 before:border-solid before:border-gray-500 before:content-[''] after:absolute after:bottom-0 after:block after:w-1/6 after:border-b-4 after:border-solid after:border-gray-900 after:content-['']"
>
  <span
    class="relative"
  >
    見出し
  </span>
</h4>
`;

exports[`HeadingLine h5タグが正しく設定されること 1`] = `
<h5
  class="undefined relative pb-4 font-bold before:absolute before:bottom-0 before:block before:w-full before:border-b-4 before:border-solid before:border-gray-500 before:content-[''] after:absolute after:bottom-0 after:block after:w-1/6 after:border-b-4 after:border-solid after:border-gray-900 after:content-['']"
>
  <span
    class="relative"
  >
    見出し
  </span>
</h5>
`;

exports[`HeadingLine h6タグが正しく設定されること 1`] = `
<h6
  class="undefined relative pb-4 font-bold before:absolute before:bottom-0 before:block before:w-full before:border-b-4 before:border-solid before:border-gray-500 before:content-[''] after:absolute after:bottom-0 after:block after:w-1/6 after:border-b-4 after:border-solid after:border-gray-900 after:content-['']"
>
  <span
    class="relative"
  >
    見出し
  </span>
</h6>
`;

exports[`HeadingLine 子要素が正しく表示されること 1`] = `
<h2
  class="undefined relative pb-4 font-bold before:absolute before:bottom-0 before:block before:w-full before:border-b-4 before:border-solid before:border-gray-500 before:content-[''] after:absolute after:bottom-0 after:block after:w-1/6 after:border-b-4 after:border-solid after:border-gray-900 after:content-['']"
>
  <span
    class="relative"
  >
    見出し
  </span>
</h2>
`;
